packages/js-moi-providers/__tests__/jsonrpc-provider.test.ts

```diff
@@ -241,4 +243,89 @@ describe("Test JsonRpcProvider Query Calls", () => {
         expect(accounts.length).toBeGreaterThanOrEqual(2)
       });
     });
+
+    describe("getNewTesseractFilter", () => {
+      it("should return the filter object containing the filter id", async () => {
+        const filter = await provider.getNewTesseractFilter();
+
+        expect(filter).toBeDefined();
+        expect(filter).toHaveProperty("id");
+        expect(typeof filter.id).toBe("string");
+      });
+    });
+
+    describe("getNewTesseractsByAccountFilter", () => {
+      it("should return a filter object containing filter id", async () => {
+        const filter = await provider.getNewTesseractsByAccountFilter(address);
+
+        expect(filter).toBeDefined();
+        expect(filter).toHaveProperty("id");
+        expect(typeof filter.id).toBe("string");
+      })
+    });
+
+    describe("getPendingInteractionFilter", () => {
+      it("should return a filter object containing filter id", async () => {
+        const filter = await provider.getPendingInteractionFilter();
+
+        expect(filter).toBeDefined();
+        expect(filter).toHaveProperty("id");
+        expect(typeof filter.id).toBe("string");
+      })
+    });
+
+    describe("removeFilter", () => {
+      it("should a return a object containing status of removal", async () => {
+        const filter = await provider.getNewTesseractsByAccountFilter(address);
+
+        const result = await provider.removeFilter(filter);
+
+        expect(result).toBeDefined();
+        expect(result).toHaveProperty("status");
+        expect(result.status).toBe(true);
+      })
+
+      it("should return a object with status false when filter is deleted or not exist", async () => {
+        const NOT_EXISTING_FILTER_ID = "678384f8-04e4-4984-9d51-44bfa5b185eb"
+        const filter: Filter = {
+          id: NOT_EXISTING_FILTER_ID
+        }
+
+        const result = await provider.removeFilter(filter);
+        
+        expect(result).toBeDefined();
+        expect(result).toHaveProperty("status");
+        expect(result.status).toBe(false);
+      })
+    });
+
+    describe("getFilterChanges", () => {
+      it("should return null when no changes made", async () => {
+        const filter = await provider.getNewTesseractFilter();
+        const result = await provider.getFilterChanges(filter);
+        expect(result).toBeNull();
+      });
+
+      it("should return an array of terreracts", async () => {
+        const nonce = await signer.getNonce();
+        const filter = await provider.getNewTesseractFilter();
+        const ixResponse = await signer.sendInteraction({
+          type: IxType.ASSET_CREATE,
+          nonce: nonce,
+          fuel_price: 1,
+          fuel_limit: 200,
+          payload: {
+              standard: AssetStandard.MAS0,
+              symbol: "TESTING",
+              supply: 1248577
+          }
+        });
+
+        await Promise.all([ixResponse.wait(), ixResponse.result()])
```

Comment:
It's not required to wait for `ixResponse.result()` to be resolved, as it internally uses the `wait` method to retrieve the receipt.

Reply:
6144a3467ccfbaf92c4f59eeb12fffd747006319